components/style/values/animated/mod.rs

```diff
@@ -88,3 +92,64 @@ where
     }
 }
 
+/// An animated value for stroke-dasharray
+///
+/// https://www.w3.org/TR/SVG/animate.html#Animatable
+#[cfg(feature = "gecko")]
+#[derive(Clone, Debug, PartialEq)]
+pub struct AnimatedStrokeDashArrayList(Vec<LengthOrPercentage>);
+
+#[cfg(feature = "gecko")]
+impl ToAnimatedValue for ComputedStrokeDashArrayList {
+    type AnimatedValue = AnimatedStrokeDashArrayList;
+
+    #[inline]
+    fn to_animated_value(self) -> Self::AnimatedValue {
+        AnimatedStrokeDashArrayList(self.0.to_animated_value())
+    }
+
+    #[inline]
+    fn from_animated_value(animated: Self::AnimatedValue) -> Self {
+        ComputedStrokeDashArrayList(ToAnimatedValue::from_animated_value(animated.0))
+    }
+}
+
+/// https://www.w3.org/TR/SVG/painting.html#StrokeDasharrayProperty
+#[cfg(feature = "gecko")]
+impl Animatable for AnimatedStrokeDashArrayList {
+    #[inline]
+    fn add_weighted(
+        &self,
+        other: &Self,
+        self_portion: f64,
+        other_portion: f64,
+    ) -> Result<Self, ()> {
```

Comment:
This was actually the new style that is being pushed from https://github.com/rust-lang-nursery/fmt-rfcs